modules/contact/src/actions/ContactAction.C

```diff
@@ -383,6 +400,14 @@ ContactAction::ContactAction(const InputParameters & params)
       paramError("mortar_dynamics",
                  "The 'mortar_dynamics' constraint option can only be used with the 'mortar' "
                  "formulation and in dynamic simulations using Newmark-beta");
+    else if (params.isParamSetByUser("quadrature"))
+      paramError("quadrature",
+                 "The 'quadrature' option can only be used with the 'mortar' formulation.");
```

Comment:
To prevent confusion, the quadrature option can also be used with `mortar_penalty`. I would call that out.